pandas/core/frame.py

```diff
@@ -374,8 +376,12 @@ def _constructor_expanddim(self):
     # ----------------------------------------------------------------------
     # Constructors
 
-    def __init__(self, data=None, index=None, columns=None, dtype=None,
-                 copy=False):
+    def __init__(self,
+                 data=None,
+                 index: Optional[Axes] = None,
```

Comment:
Ultimately deferring to @jreback but I actually don't like Optional here; I think it hurts readability without adding a lot of value